src/bin/pg_autoctl/pgctl.c

```diff
@@ -920,6 +937,156 @@ pg_ctl_start(const char *pg_ctl,
 }
 
 
+/*
+ * pg_log_startup logs the PGDATA/startup.log file contents so that our users
+ * have enough information about why Postgres failed to start when that
+ * happens.
+ */
+bool
+pg_log_startup(const char *pgdata, int logLevel)
+{
+	char pgLogDirPath[MAXPGPATH] = { 0 };
+
+	char pgStartupPath[MAXPGPATH] = { 0 };
+	char *fileContents;
+	long fileSize;
+
+	/* logLevel to use when introducing which file path logs come from */
+	int pathLogLevel = logLevel <= LOG_DEBUG ? LOG_DEBUG : LOG_WARN;
+
+	/* prepare startup.log file in PGDATA */
+	join_path_components(pgStartupPath, pgdata, "startup.log");
+
+	if (read_file(pgStartupPath, &fileContents, &fileSize) && fileSize > 0)
```

Comment:
That's how Postgres work, as far as I understand. We certainly want to avoid losing those startup logs, because in some cases those are very useful.

Reply:
I agree that it is very useful. 

My point was, even if we're mostly interested in the latest (or maybe few of the latest) logs, we might be printing thousands of restart logs that are in `startup.log`. We can definitely live with that, though users might be confused seeing old logs -- which may even contain unrelated errors to our current error situation.

So, if there is an easy way to distinguish the latest startup logs, say the last 5 ones, it might be more interesting to see I guess.

Reply:
We actually replace the file content at each Postgres start, because of the following code in `pg_ctl_postgres` where we open the file in write mode rather than in append mode:

``` c
	/* we want to redirect the output to logfile */
	logFileDescriptor = open(logfile, FOPEN_FLAGS_W, 0644);
```

Did you actually see `startup.log` content from multiple Postgres starts in your testing?